From b1420ee264d6c462454919492f53a506d7bce0e1 Mon Sep 17 00:00:00 2001
From: Tim Waugh <twaugh@redhat.com>
Date: Tue, 2 Aug 2011 15:43:11 +0100
Subject: [PATCH 4/4] Removed PackageKit client code in missingPackagesAndExecutables (bug #726938).

The PackageKit Python client module is hard-coded to set a default
GLib main loop on import, which means it cannot be imported by a
module that might be used by an application using a different main
loop implementation, such as KDE's printer-applet.
---
 cupshelpers/cupshelpers.py |   24 +-----------------------
 1 files changed, 1 insertions(+), 23 deletions(-)

diff --git a/cupshelpers/cupshelpers.py b/cupshelpers/cupshelpers.py
index 3b95e2f..bf6eda2 100755
--- a/cupshelpers/cupshelpers.py
+++ b/cupshelpers/cupshelpers.py
@@ -22,11 +22,6 @@
 import cups, pprint, os, tempfile, re, string
 import locale
 from . import _debugprint
-try:
-    import packagekit.client, packagekit.enums
-    HAVE_PACKAGEKIT=True
-except ImportError:
-    HAVE_PACKAGEKIT=False
 
 class Printer:
     _flags_blacklist = ["options", "local"]
@@ -833,24 +828,7 @@ def missingPackagesAndExecutables(ppd):
     missing executables
     """
     executables = missingExecutables(ppd)
-    packages = []
-    if executables and HAVE_PACKAGEKIT:
-        unresolved_executables = []
-        client = packagekit.client.PackageKitClient ()
-        for executable in executables:
-            if not executable.startswith ("/"):
-                executable = "/usr/bin/" + executable
-
-            result = client.search_file ([executable],
-                                         packagekit.enums.FILTER_NOT_INSTALLED)
-            if result:
-                packages.extend (set (map (lambda x: x.name, result)))
-            else:
-                unresolved_executables.append (executable)
-
-        executables = unresolved_executables
-        
-    return (packages, executables)
+    return ([], executables)
 
 def _main():
     c = cups.Connection()
-- 
1.7.3.4

